hathor/transaction/base_transaction.py

```diff
@@ -742,6 +743,51 @@ def _update_feature_activation_bit_counts(self) -> None:
         # This method lazily calculates and stores the value in metadata
         self.get_feature_activation_bit_counts()
 
+    def _update_closest_block_metadata(self) -> None:
+        """
+        Set the tx's closest_block metadata.
+        For blocks, it's always None. For Transactions, it's the Block with the greatest height that is a direct
+        or indirect dependency (ancestor) of the transaction, including both funds and confirmation DAGs.
+        It's calculated by propagating the metadata forward in the DAG,
+        and it's used by Feature Activation for Transactions.
+        """
+        from hathor.transaction import Block, Transaction
+        if isinstance(self, Block):
+            return
+        assert isinstance(self, Transaction)
+        assert self.storage is not None
+        metadata = self.get_metadata()
```

Comment:
Check if `metadata.closest_block is not None` and returns if true.

Reply:
This has been removed.

Reply:
Thanks!